queries.sql

```diff
@@ -8,3 +8,41 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon'; 
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+ROLLBACK;
+
+BEGIN;
+SAVEPOINT weight_save;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO weight_kg;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg <0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) FROM animals;
+
+
+SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
```

Comment:
- [ OPTIONAL ] I think here the question is Who escapes the most, neutered or not neutered animals so it would be better if you use Max instead od SUM 🚀😊🙏🏻🔥

Reply:
@Reem-lab Thank you for the comment I will work on it